js/plugin/popup-dialog.ts

```diff
@@ -210,51 +179,240 @@ export class BrowserModPopup extends LitElement {
+    // Check viewport margin on window resize
+    this._resizeObserver = new ResizeObserver(() => {
+      this._checkViewportMargin();
+    });
+
+    // Observe window size changes
+    this._resizeObserver.observe(document.body);
+
+    // Initial check
+    this._checkViewportMargin();
+  }
+
+  private _checkViewportMargin() {
+    const MARGIN_THRESHOLD = 50; // px
+    const windowWidth = window.innerWidth;
+
+    // Get the dialog element to read its actual styles
+    const dialogElement = this.shadowRoot?.querySelector('ha-dialog') as HTMLElement;
+    if (!dialogElement) return;
+
+    const computedStyle = getComputedStyle(dialogElement);
+    let dialogMinWidth = 580; // default
+    let dialogMaxWidth = 580; // default
+
+    // Parse min/max width from CSS variables
+    const minWidthStr = computedStyle.getPropertyValue('--mdc-dialog-min-width').trim();
+    const maxWidthStr = computedStyle.getPropertyValue('--mdc-dialog-max-width').trim();
+
+    // Only parse if not in narrow mode (not 100vw)
+    if (minWidthStr && !minWidthStr.includes('100vw')) {
+      const parsed = parseFloat(minWidthStr);
+      if (!isNaN(parsed)) {
+        dialogMinWidth = minWidthStr.includes('vw') ? (parsed / 100) * windowWidth : parsed;
+      }
+    }
+    if (maxWidthStr && !maxWidthStr.includes('100vw')) {
+      const parsed = parseFloat(maxWidthStr);
+      if (!isNaN(parsed)) {
+        dialogMaxWidth = maxWidthStr.includes('vw') ? (parsed / 100) * windowWidth : parsed;
+      }
+    }
+
+    // Use the configured width
+    const dialogWidth = Math.max(dialogMinWidth, dialogMaxWidth);
+
+    // Calculate horizontal margin (space on both sides combined)
+    const totalHorizontalMargin = windowWidth - dialogWidth;
+
+    // Set narrow attribute if total margin is too small
+    // This means there's less than MARGIN_THRESHOLD pixels on each side
+    const shouldBeNarrow = totalHorizontalMargin <= (MARGIN_THRESHOLD * 2);
+
+    if (this._narrow !== shouldBeNarrow) {
+      this._narrow = shouldBeNarrow;
+    }
+  }
+
+  private _build_card = async (config: any): Promise<void> => {
+    try {
+      const helpers = await window.loadCardHelpers();
+      const card = await helpers.createCardElement(config);
+      card.hass = window.browser_mod.hass;
+      provideHass(card);
+      this.content = card;
+
+      if (!customElements.get(card.localName)) {
+        customElements.whenDefined(card.localName).then(() => {
+          this._build_card(config);
+        });
+      }
 
-  async _build_card(config) {
-    const helpers = await window.loadCardHelpers();
-    const card = await helpers.createCardElement(config);
-    card.hass = window.browser_mod.hass;
-    provideHass(card);
-    this.content = card;
-    if (!customElements.get(card.localName)) {
-      customElements.whenDefined(card.localName).then(() => {
+      card.addEventListener("ll-rebuild", () => {
         this._build_card(config);
       });
+    } catch (error) {
```

Comment:
No. This is not required. The error is caught by Frontend which will display error title and details in preview mode, and 'Configuration error' otherwise.